core/src/main/scala/org/broadinstitute/workbench/hamm/pricing/model.scala

```diff
@@ -1,10 +1,91 @@
 package org.broadinstitute.workbench.hamm.pricing
 
+import cats.data.NonEmptyList
 import io.circe.Decoder
+import org.broadinstitute.workbench.hamm.Region
+import org.broadinstitute.workbench.hamm.pricing.UsageType.{COMMIT1YR, ONDEMAND, PREEMPTIBLE}
 
-final case class CpuCost(asDouble: Double) extends AnyVal
-final case class RamCost(asDouble: Double) extends AnyVal
 
-final case class ComputeCost(cpuCost: CpuCost, ramCost: RamCost) {
-  val totalCost = cpuCost.asDouble + ramCost.asDouble //TODO: update this
+sealed trait UsageType {
+  def asString: String
+  def asDescriptionString: String
 }
+
+object UsageType {
+  private final val PREEMPTIBLE = "Preemptible"
+  private final val ONDEMAND = "OnDemand"
+  private final val COMMIT1YR = "Commit1Yr"
+
+  private final val PREEMPTIBLE_DESCRIPTION_STRING = "Preemptible"
+  private final val ONDEMAND_DESCRIPTION_STRING = ""
+  private final val COMMIT1YR_DESCRIPTION_STRING = "Commitment v1:"
+
+  val stringToUsageType = Map(
+    PREEMPTIBLE -> Preemptible,
+    ONDEMAND -> OnDemand,
+    COMMIT1YR -> Commit1Yr
+  )
+
+  case object Preemptible extends UsageType {
+    def asString = PREEMPTIBLE
+    def asDescriptionString: String = PREEMPTIBLE_DESCRIPTION_STRING
+  }
+
+  case object OnDemand extends UsageType {
+    def asString = ONDEMAND
+    def asDescriptionString: String = ONDEMAND_DESCRIPTION_STRING
+  }
+
+  case object Commit1Yr extends UsageType {
+    def asString = COMMIT1YR
+    def asDescriptionString: String = COMMIT1YR_DESCRIPTION_STRING
+  }
+}
+
+sealed trait ResourceFamily {
+  def asString: String
+  def asDecodingFailureMessage: String
```

Comment:
what's `asDecodingFailureMessage` for?

Reply:
vestige of something I was trying to do before, I'll remove it